tests/functional/aws-node-sdk/test/object/copyPart.js

```diff
@@ -580,69 +580,86 @@ describe('Object Part Copy', () => {
             });
 
             it('should not corrupt object if overwriting an existing part by copying a part ' +
-            'while the MPU is being completed', () => {
-                // AWS response etag for this completed MPU
-                const finalObjETag = '"db77ebbae9e9f5a244a26b86193ad818-1"';
-                process.stdout.write('Putting first part in MPU test');
-                return s3.uploadPartCopy({ Bucket: destBucketName,
-                    Key: destObjName,
-                    CopySource: `${sourceBucketName}/${sourceObjName}`,
-                    PartNumber: 1,
-                    UploadId: uploadId,
-                }).promise().then(res => {
+            'while the MPU is being completed', async () => {
+                    const finalObjETag = '"db77ebbae9e9f5a244a26b86193ad818-1"';
+                    process.stdout.write('Putting first part in MPU test"');
+                    const randomDestObjName = `copycatobject${Math.floor(Math.random() * 100000)}`;
+
+                    const initiateRes = await s3
+                      .createMultipartUpload({
```

Comment:
this call was not made before 🤔
how could it work? was it done in a fixture?

Reply:
@francoisferrand it was done in the before. But we use the same key for several tests that can, IMO, cause collision.  That's why I decided to move to it's own key / multipart upload